app/javascript/components/shared_components/Footer.jsx

```diff
@@ -17,19 +17,21 @@
 import React from 'react';
 import { useTranslation } from 'react-i18next';
 import { Container } from 'react-bootstrap';
+import PropTypes from 'prop-types';
 import useEnv from '../../hooks/queries/env/useEnv';
 import useSiteSetting from '../../hooks/queries/site_settings/useSiteSetting';
 
-export default function Footer() {
+export default function Footer({ currentUser = null }) {
```

Comment:
You don't need to get `currentUser` through props. Props are only needed if you need to track their state across multiple components. In this case, we can just get current user directly in the Footer component